mobile/native/src/trade/order/handler.rs

```diff
@@ -1,46 +1,41 @@
+use crate::config;
 use crate::db;
 use crate::event;
 use crate::event::EventInternal;
-use crate::ln_dlc;
 use crate::trade::order::FailureReason;
 use crate::trade::order::Order;
 use crate::trade::order::OrderState;
-use crate::trade::position;
 use anyhow::bail;
 use anyhow::Context;
 use anyhow::Result;
-use std::time::Duration;
-use trade::ContractSymbol;
-use trade::TradeParams;
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
 use uuid::Uuid;
 
 pub async fn submit_order(order: Order) -> Result<()> {
     db::insert_order(order)?;
 
-    ui_update(order);
-
-    // TODO: remove this dummy
-    let dummy_trade_params = TradeParams {
-        pubkey: ln_dlc::get_node_info()?.pubkey,
-        // We set this to our pubkey as well for simplicity until we receive a match from the
-        // orderbook
-        pubkey_counterparty: ln_dlc::get_node_info()?.pubkey,
-        // We set this to our order as well for simplicity until we receive a match from the
-        // orderbook
-        order_id: order.id,
-        order_id_counterparty: "the orderbook will know".to_string(),
-        contract_symbol: ContractSymbol::BtcUsd,
-        leverage: order.leverage,
-        leverage_counterparty: 2.0,
-        quantity: order.quantity,
-        execution_price: 55_000.0,
-        // in 24h
-        expiry: Duration::from_secs(60 * 60 * 24),
-        oracle_pk: ln_dlc::get_oracle_pubkey()?,
+    // todo: this model should include the contract_symbol
+    let new_order = trade::NewOrder {
```

Comment:
We will need to know if this is a market order or a limit order

Reply:
This is how the `NewOrder` is represented on the API for the app; I think this should be complete. Likely we will need a similar representation when sending an order to the orderbook.

```
#[frb]
#[derive(Debug, Clone)]
pub struct NewOrder {
    #[frb(non_final)]
    pub leverage: f64,
    #[frb(non_final)]
    pub quantity: f64,
    #[frb(non_final)]
    pub contract_symbol: ContractSymbol,
    #[frb(non_final)]
    pub direction: Direction,
    // Box needed for complex enum, otherwise generated Rust code complains about Default impl
    // missing
    #[frb(non_final)]
    pub order_type: Box<OrderType>,
}

```

Reply:
@da-kami why is the quantity of type `f64`? 

and I guess the `order_type` is `limit` or `market`?

Reply:
what about the `pub_key` of the trader?

Reply:
True, the pubkey of the trader should be added. On the app it's our own so we know it internally and don't send it to the API; for the orderbook we will have to add it.